### PR TITLE
Convert JUnit 4 to 5

### PR DESCRIPTION
**Pull request by corneil**
Convert Hamcrest to AssertJ

### CODE REVIEW

File: spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/DefaultContainerImageMetadataResolverTest.java [hunk 1]
```diff
@@ -23,8 +23,9 @@
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import org.junit.Before;
-import org.junit.Test;
+import org.junit.jupiter.api.BeforeAll;
```
Comment (cppwfs) [inline spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/DefaultContainerImageMetadataResolverTest.java:26]: BeforeAll import can be removed

File: spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/authorization/DockerConfigJsonSecretToRegistryConfigurationConverterTest.java [hunk 2]
```diff
@@ -21,8 +21,9 @@
 import java.util.HashMap;
 import java.util.Map;
 
-import org.junit.Before;
-import org.junit.Test;
+import org.junit.jupiter.api.BeforeAll;
```
Comment (cppwfs) [inline spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/authorization/DockerConfigJsonSecretToRegistryConfigurationConverterTest.java:24]: BeforeAll import can be removed

**Review by cppwfs — CHANGES_REQUESTED** [review]
Looks good.  Had 3 comments.   Did notice that these changes are causing compilation errors with unmigrated code in Spring Cloud Dataflow Commons.

File: pom.xml [hunk 3]
```diff
@@ -147,4 +153,36 @@
 			</plugin>
 		</plugins>
 	</build>
+	<profiles>
```
[the remark this answers is not in the document]
Comment (corneil) [reply pom.xml:156]: I have removed it

File: spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/DefaultContainerImageMetadataResolverTest.java (hunk 1 above)
Comment (corneil) [reply spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/DefaultContainerImageMetadataResolverTest.java:26]: Noted

File: spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/authorization/DockerConfigJsonSecretToRegistryConfigurationConverterTest.java (hunk 2 above)
Comment (corneil) [reply spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/authorization/DockerConfigJsonSecretToRegistryConfigurationConverterTest.java:24]: Noted

File: spring-cloud-dataflow-core/src/test/java/org/springframework/cloud/dataflow/core/StreamDefinitionTests.java [hunk 4]
```diff
@@ -19,17 +19,14 @@
...
 import org.springframework.cloud.dataflow.core.dsl.ParseException;
 import org.springframework.cloud.dataflow.core.dsl.StreamParser;
 
-import static org.hamcrest.Matchers.containsString;
-import static org.hamcrest.Matchers.is;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertThat;
-import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (onobc) [inline spring-cloud-dataflow-core/src/test/java/org/springframework/cloud/dataflow/core/StreamDefinitionTests.java:28]: Please remove all wildcard imports (its in a few locations in this PR).

**Review by onobc — CHANGES_REQUESTED** [review]
@corneil thanks for this migration! Nice work. I have added quite a few smaller comments/suggestions but the main concern I have is the dep. mgmt. of Junit and Mockito which we should not need (comments have more details about this). 

Thanks

File: rewrite.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (corneil) [reply rewrite.sh:1]: I believe this script is valuable for future use by adding recipes as needed.

File: spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml [hunk 6]
```diff
@@ -38,13 +38,56 @@
...
+				<exclusions>
+					<exclusion>
+						<groupId>org.junit.jupiter</groupId>
+						<artifactId>junit-jupiter-api</artifactId>
+					</exclusion>
+				</exclusions>
+			</dependency>
 			<dependency>
 				<groupId>net.minidev</groupId>
 				<artifactId>json-smart</artifactId>
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml:93]: fixed

File: spring-cloud-dataflow-container-registry/src/test/java/org/springframework/cloud/dataflow/container/registry/ContainerImageParserTests.java [hunk 7]
```diff
@@ -105,13 +105,13 @@ public void testParseWithDefaults() {
 
 	@Test
 	public void testInvalidRegistryHostName() {
-		assertThrows(IllegalArgumentException.class, () ->
+		assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() ->
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-container-registry/src/test/java/org/springframework/cloud/dataflow/container/registry/ContainerImageParserTests.java:108]: Nice

**Comment by corneil** [comment]
Allow for JUnit 4 tests to execute until we are done.

File: spring-cloud-skipper/spring-cloud-skipper-shell/pom.xml [hunk 8]
```diff
@@ -27,6 +27,11 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>junit</groupId>
+			<artifactId>junit</artifactId>
+			<scope>test</scope>
+		</dependency>
 		<dependency>
```
Comment (cppwfs) [inline spring-cloud-skipper/spring-cloud-skipper-shell/pom.xml:35]: Currently fails in build with ` spring-cloud-skipper-shell: No tests to run`. 
We may need to add this temporarily :

```
				<configuration>
					<skipTests>true</skipTests>
				</configuration>
```

**Review by cppwfs — CHANGES_REQUESTED** [review]
Thanks for doing this!   Once we get a successful build, then we can proceed to the next modules.

File: rewrite.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (onobc) [reply rewrite.sh:1]: How so?

File: spring-cloud-common-security-config/spring-cloud-common-security-config-core/pom.xml [hunk 9]
```diff
@@ -29,5 +29,10 @@
 			<artifactId>javax.servlet-api</artifactId>
 			<scope>provided</scope>
 		</dependency>
+		<dependency>
```
Comment (onobc) [inline spring-cloud-common-security-config/spring-cloud-common-security-config-core/pom.xml:32]: If this is required, we could have just added it where it was before at the 2 roots (scdf-parent and scdf-build-dependencies). I am guessing this will come out in the next phase of the PR though.

File: spring-cloud-dataflow-build/pom.xml [hunk 10]
```diff
@@ -203,6 +187,9 @@
 					<configuration>
 						<groups>${groups}</groups>
 						<excludedGroups>${excludedGroups}</excludedGroups>
+						<excludes>
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-build/pom.xml:190]: I am still having an issue with failsafe-plugin so this may change.

File: rewrite.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (corneil) [reply rewrite.sh:1]: OK

File: spring-cloud-dataflow-build/pom.xml [hunk 11]
```diff
@@ -104,22 +104,6 @@
 	</dependencyManagement>
 	<dependencies>
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-build/pom.xml:105]: I had to add back junit 4 dependencies until we are done with removing

File: spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/configuration/metadata/BootApplicationConfigurationMetadataResolverTests.java [hunk 12]
```diff
@@ -130,31 +125,31 @@ public void deprecatedErrorPropertiesShouldNotBeVisible() {
...
 	public void shouldReturnPortMappingProperties() {
 		Map<String, Set<String>> portNames = resolver.listPortNames(new ClassPathResource("apps/filter-processor", getClass()));
-		assertThat(portNames.size(), is(2));
-		assertThat(portNames.get("inbound").size(), is(3));
-		assertThat(portNames.get("inbound"), containsInAnyOrder("in1", "in2", "in3"));
-		assertThat(portNames.get("outbound").size(), is(2));
-		assertThat(portNames.get("outbound"), containsInAnyOrder("out1", "out2"));
+		assertThat(portNames).hasSize(2);
+		assertThat(portNames.get("inbound")).hasSize(3);
+		assertThat(portNames.get("inbound")).containsExactlyInAnyOrder("in1", "in2", "in3");
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/configuration/metadata/BootApplicationConfigurationMetadataResolverTests.java:136]: I have cleaned up those cases

File: spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/authorization/S3SignedRedirectRequestServerResource.java [hunk 13]
```diff
@@ -28,20 +31,17 @@
 /**
  * @author Adam J. Weigold
  */
-public class S3SignedRedirectRequestServerResource extends ExternalResource {
+public class S3SignedRedirectRequestServerResource implements BeforeEachCallback, AfterEachCallback {
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/authorization/S3SignedRedirectRequestServerResource.java:34]: Tx

File: spring-cloud-dataflow-core/src/test/java/org/springframework/cloud/dataflow/core/ArgumentSanitizerTest.java [hunk 14]
```diff
@@ -34,16 +35,16 @@ public class ArgumentSanitizerTest {
...
 	public void before() {
 		sanitizer = new ArgumentSanitizer();
 	}
 
 	@Test
 	public void testSanitizeProperties() {
 		for (String key : keys) {
-			Assert.assertEquals("--" + key + "=******", sanitizer.sanitize("--" + key + "=foo"));
-			Assert.assertEquals("******", sanitizer.sanitize(key, "bar"));
+			assertEquals("--" + key + "=******", sanitizer.sanitize("--" + key + "=foo"));
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-core/src/test/java/org/springframework/cloud/dataflow/core/ArgumentSanitizerTest.java:46]: Done

File: spring-cloud-skipper/spring-cloud-skipper-shell/pom.xml (hunk 8 above)
Comment (corneil) [reply spring-cloud-skipper/spring-cloud-skipper-shell/pom.xml:35]: failNoTests=false should to the job

File: spring-cloud-common-security-config/spring-cloud-common-security-config-core/pom.xml (hunk 9 above)
Comment (corneil) [reply spring-cloud-common-security-config/spring-cloud-common-security-config-core/pom.xml:32]: Removing all junit 4 at the end will be easy. If everything builds it is easier to review.

File: spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml [hunk 15]
```diff
@@ -38,13 +38,56 @@
...
+		<junit.version>4.13.2</junit.version>
+		<junit-jupiter.version>5.9.3</junit-jupiter.version>
+		<junit-platform.version>1.9.3</junit-platform.version>
+		<mockito.version>4.11.0</mockito.version>
 		<logback.version>1.2.13</logback.version>
 		<json-path.version>2.9.0</json-path.version>
 	</properties>
 	<dependencyManagement>
 		<dependencies>
+			<dependency>
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml:50]: We are bumping to latest that should work with Boot 2.7.x

File: spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml [hunk 16]
```diff
@@ -38,13 +38,56 @@
 		<java-cfenv.version>2.3.0</java-cfenv.version>
 		<spring-cloud-services-starter-config-client.version>3.5.4</spring-cloud-services-starter-config-client.version>
 		<kubernetes-fabric8-client.version>5.12.4</kubernetes-fabric8-client.version>
-		<junit.version>4.13.1</junit.version>
-		<junit-jupiter.version>5.9.2</junit-jupiter.version>
+		<junit.version>4.13.2</junit.version>
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml:41]: This is main not main-3

File: spring-cloud-dataflow-composed-task-runner/pom.xml [hunk 17]
```diff
@@ -94,6 +94,17 @@
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-test</artifactId>
+			<exclusions>
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-composed-task-runner/pom.xml:97]: There are cases where junit and junit-jupiter is transitive without a property then Spring Boot will not help. I'm trying to get to later version

File: spring-cloud-dataflow-parent/pom.xml [hunk 18]
```diff
@@ -56,14 +56,67 @@
 		<logback.version>1.2.13</logback.version>
 		<json-path.version>2.9.0</json-path.version>
 		<postgresql.version>42.7.2</postgresql.version>
+		<junit-jupiter.version>5.9.3</junit-jupiter.version>
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-parent/pom.xml:59]: This is boot 2.7.18 not 3.x

File: spring-cloud-dataflow-build/pom.xml [hunk 19]
```diff
@@ -234,8 +221,8 @@
 				<plugin>
 					<groupId>org.apache.maven.plugins</groupId>
 					<artifactId>maven-surefire-plugin</artifactId>
-					<version>${maven-surefire-plugin.version}</version>
 					<configuration>
+						<failIfNoTests>true</failIfNoTests>
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-build/pom.xml:225]: When we have all modules converted we can pull all module specific cases.

File: spring-cloud-dataflow-composed-task-runner/src/test/java/org/springframework/cloud/dataflow/composedtaskrunner/support/OnOAuth2ClientCredentialsEnabledTests.java [hunk 20]
```diff
@@ -43,19 +43,19 @@ public void teardown() {
 	@Test
 	public void noPropertySet() throws Exception {
 		this.context = load(Config.class);
-		org.hamcrest.MatcherAssert.assertThat(context.containsBean("myBean"), equalTo(false));
+		assertThat(context.containsBean("myBean")).isEqualTo(false);
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-composed-task-runner/src/test/java/org/springframework/cloud/dataflow/composedtaskrunner/support/OnOAuth2ClientCredentialsEnabledTests.java:46]: That was open-rewrite.

File: spring-cloud-dataflow-composed-task-runner/pom.xml (hunk 17 above)
[the remark this answers is not in the document]
Comment (onobc) [reply spring-cloud-dataflow-composed-task-runner/pom.xml:97]: Why are you trying to get to later version? Is there a CVE? Is there some feature needed?

File: spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml (hunk 16 above)
[the remark this answers is not in the document]
Comment (onobc) [reply spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml:41]: What do we need from the very latest that is not in the version used by Boot? The risk / reward for having to manage these dependencies ourselves is not there IMO. If we let Boot manage this all of this dependency info comes out of here.

[the remark this answers is not in the document]
Comment (onobc) [reply spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml:41]: I meant 2.7.x which is on https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-dependencies/2.7.15 which is on 5.8.2

File: spring-cloud-dataflow-parent/pom.xml (hunk 18 above)
[the remark this answers is not in the document]
Comment (onobc) [reply spring-cloud-dataflow-parent/pom.xml:59]: Yeh, I had the wrong links and version but it still applies w/ SB 2.7.x which is on Junit 5.8.x

File: spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml [hunk 21]
```diff
@@ -39,13 +39,24 @@
...
+		<mockito.version>4.11.0</mockito.version>
 		<logback.version>1.2.13</logback.version>
 		<json-path.version>2.9.0</json-path.version>
+		<guava.version>32.1.3-jre</guava.version>
 	</properties>
 	<dependencyManagement>
 		<dependencies>
+			<dependency>
+				<groupId>com.google.guava</groupId>
```
Comment (onobc) [inline spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml:51]: Why is this needed?

File: spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml [hunk 22]
```diff
@@ -39,13 +39,24 @@
...
+				<artifactId>guava</artifactId>
+				<version>${guava.version}</version>
+			</dependency>
+			<dependency>
+				<groupId>junit</groupId>
+				<artifactId>junit</artifactId>
+				<version>4.13.2</version>
```
Comment (onobc) [inline spring-cloud-dataflow-build/spring-cloud-dataflow-build-dependencies/pom.xml:58]: All we need to do is set `junit.version` and the SB dep. mgmt will do the right thing

File: spring-cloud-dataflow-build/spring-cloud-dataflow-build-tools/pom.xml [hunk 23]
```diff
@@ -28,6 +28,13 @@
 					<target>1.8</target>
 				</configuration>
 			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
```
Comment (onobc) [inline spring-cloud-dataflow-build/spring-cloud-dataflow-build-tools/pom.xml:32]: Looks like this had to be added to every single pom.xml?
